server/github.go

```diff
@@ -168,3 +171,54 @@ func (s *Server) checkUserPermission(user, repoOwner string) bool {
 
 	return true
 }
+
+func (s *Server) createRefWithPrefixFromPr(pr *model.PullRequest, prefix string) {
+	client := NewGithubClient(s.Config.GithubAccessToken)
+	_, _, err := client.Git.CreateRef(
+		context.Background(),
+		pr.RepoOwner,
+		pr.RepoName,
+		&github.Reference{
+			Ref: github.String("refs/heads/" + prefix + strconv.Itoa(pr.Number)),
+			Object: &github.GitObject{
+				SHA: github.String(pr.Sha),
+			},
+		})
+
+	if err != nil {
+		mlog.Error("Error creating reference", mlog.Err(err))
+	}
+}
+
+func (s *Server) deleteRefsWithPrefixWhereCombinedStateEqualsSuccess(repoOwner string, repoName string, prefix string) {
+	client := NewGithubClient(s.Config.GithubAccessToken)
+
+	branches, _, err := client.Repositories.ListBranches(context.Background(), repoOwner, repoName, nil)
+	if err != nil {
+		mlog.Error("Error listing branches", mlog.Err(err))
+	}
+
+	regexBranchPrefix := regexp.MustCompile(`^` + prefix + `[0-9]+$`)
+
+	for _, branch := range branches {
```

Comment:
this will delete other branches that might be running and needed. can we delete only the requested one?

Reply:
Fixed this one and will only delete the requested one. Way better now, thank you.